main.go

```diff
@@ -14,47 +16,220 @@ import (
 	"sync"
 	"time"
 
-	flag "github.com/spf13/pflag"
+	"github.com/olekukonko/tablewriter"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// possible values for the output that will be printed to the terminal
+const (
+	standart format = iota
+	fancy
+	simple
+)
+
+var (
+	srv         *string = flag.String("srv", "_service._proto.exmaple.com", "the srv record name to be used to look up IP addresses and port")
+	listenAddr  *string = flag.String("listen-address", ":3000", "The service will be listening to that address with port\ne.g. 172.0.0.1:3000")
+	metricsAddr *string = flag.String("metrics-address", ":9090", "The metrics server will be listening to that address with port\ne.g. 172.0.0.1:9090")
+)
+
+var (
+	httpVectorClient = http.Client{
+		Timeout: 10 * time.Second,
+	}
+	httpPingClient = http.Client{
+		Timeout: 5 * time.Second,
+	}
+)
+
+var (
+	fullScanCounter = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "fullscan_counter",
```

Comment:
The convention is for counter variables to be plural and have the suffix `_total`
```suggestion
			Name: "fullscans_total",
```

1.